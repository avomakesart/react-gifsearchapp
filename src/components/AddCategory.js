import React, { useState } from 'react'
import PropTypes from 'prop-types'

export default function AddCategory({ setCategories }) {
  const [inputValue, setInputValue] = useState('')

  const handleInputChange = (e) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e) => {
    e.preventDefault()

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats])
      setInputValue('')
    }
  }

  return (
    <>
      <h3>Agregar Categoria</h3>
      <form onSubmit={handleSubmit}>
        <input type="text" value={inputValue} onChange={handleInputChange} placeholder="Realiza una busqueda..."/>
      </form>
    </>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}
