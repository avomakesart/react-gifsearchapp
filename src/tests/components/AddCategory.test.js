import React from 'react'
import '@testing-library/jest-dom'
import AddCategory from '../../components/AddCategory'
import { shallow } from 'enzyme'

describe('Testing <AddCategory />', () => {
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('Should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should change the text input', () => {
    const input = wrapper.find('input')
    const value = 'Hello world'

    input.simulate('change', { target: { value } })
  })

  test('should not post the info with submit fn', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} })

    expect(setCategories).not.toHaveBeenCalled()
  })

  test('should call setCategories and clean the text input', async () => {
    const input = wrapper.find('input')
    const value = 'Hello world'

    input.simulate('change', { target: { value } })

    wrapper.find('form').simulate('submit', { preventDefault() {} })

    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))

    expect(input.prop('value')).toBe('')
  })
})
