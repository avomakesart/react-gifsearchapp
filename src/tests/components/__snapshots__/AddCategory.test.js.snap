// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AddCategory /> Should show the component correctly 1`] = `
<Fragment>
  <h3>
    Agregar Categoria
  </h3>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Realiza una busqueda..."
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
